src/server.cpp

```diff
@@ -593,6 +595,28 @@ void Server::AsyncRunStep(bool initial_step)
 	}
 
 	/*
+		Listen to the admin chat, if available
+	*/
+	if (m_admin_chat) {
+		if (!m_admin_chat->command_queue.empty()) {
+			MutexAutoLock lock(m_env_mutex);
+			while (!m_admin_chat->command_queue.empty()) {
+				ChatEvent *evt = m_admin_chat->command_queue.pop_frontNoEx();
+				if (evt->type == CET_NICK_ADD) {
+					// The terminal informed us of its nick choice
+					m_admin_nick = ((ChatEventNick *)evt)->nick;
+				} else {
+					assert(evt->type == CET_CHAT);
+					handleAdminChat((ChatEventChat *)evt);
+				}
+				delete evt;
+			}
+		}
+	}
+	m_admin_chat->outgoing_queue.push_back(
```

Comment:
I am getting a segfault here (with m_admin_chat == NULL). I think this should be inside the if block.